tests/test_flows.py

```diff
@@ -707,30 +707,28 @@ def test_link_metric_query(self):
         query = self.check_cmdu_type_single("link metric query", 0x0005,
                                             env.controller.mac, env.agents[0].mac,
                                             mid)
+        assert query
 
-        if query:
-            query_tlv = self.check_cmdu_has_tlv_single(query, 0x08)
+        query_tlv = self.check_cmdu_has_tlv_single(query, 0x08)
+        assert query_tlv
 
-        if query and query_tlv:
-            debug("Checking query type and queried metrics are correct")
-            try:
-                if int(query_tlv.link_metric_query_type) != 0x00:
-                    self.fail("Query type is not 'All neighbors'")
-            except (AttributeError, ValueError):
-                self.fail("Query type is not recognized")
-            try:
-                if int(query_tlv.link_metrics_requested) != 0x02:
-                    self.fail("Metrics for both Tx and Rx is not requested")
-            except (AttributeError, ValueError):
-                self.fail("Requested metrics are not recognized")
+        debug("Checking query type and queried metrics are correct")
+        assert query_tlv.link_metric_query_type, "Query type is not recognized"
+        assert int(query_tlv.link_metric_query_type) == 0x00, \
+            "Query type is not 'All neighbors'"
```

Comment:
I don't think we should use asserts everywhere. It's better if we can continue the test, so assert isn't always good. And in this case I actually think the original code is easier to read.